### ISSUE TITLE
Non-tools builds fail linking on Linux with ld.gold

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot#31197]
**Godot version:**
3.2 master (1d5ae6d) after #24086 

**OS/device including version:**
Mageia 7 x86_64
```
$ gcc --version | head -n 1
gcc (Mageia 8.3.1-0.20190524.1.mga7) 8.3.1 20190524
$ ld.gold --version | head -n 1
GNU gold (GNU Binutils 2.32) 1.16
```

**Issue description:**
When compiling with `ld.gold` as linker, the linking step fails with:
```
Linking Program        ==> bin/godot.x11.opt.64
/bin/ld.gold: error: platform/x11/pck_embed.ld:10:8: syntax error, unexpected STRING
/bin/ld.gold: fatal error: unable to parse script file platform/x11/pck_embed.ld
collect2: error: ld returned 1 exit status
```

It works fine with `ld` (GNU ld (GNU Binutils) 2.32).

CC @RandomShaper 

**Steps to reproduce:**

- Make sure you have `ld.gold` installed and in PATH
- `scons p=x11 LINKFLAGS=-fuse-ld=gold tools=no target=release`

**Comment by akien-mga** [comment]
For the reference, it links fine with LLD (`use_llvm=yes use_lld=yes`).

**Comment by creikey** [comment]
This stackoverflow answer seems to have some pointers to using gold: https://stackoverflow.com/questions/3476093/replacing-ld-with-gold-any-experience

**Comment by Anutrix** [comment]
> The initial design of gold meant to not support any linker scripts. Then
> they end up supporting some cases, but "INSERT BEFORE .data" is not one of
> them if I remember correctly.

Source: https://github.com/clang-omp/clang_trunk/issues/30#issuecomment-197918536
Maybe the build is failing because gold just doesn't support BEFORE and AFTER.

**Comment by KoBeWi** [comment]
Can anyone still reproduce this bug in Godot 3.2.3 or any later release?

**Comment by akien-mga** [comment]
Yes, the same method is still used with a linker script for the PCK bundling option, which is incompatible with `ld.gold`.

**Comment by RandomShaper** [comment]
I hope I can allocate some time soon to rework the PCK embedding so we can remove the current fragile approach.

**Comment by akien-mga** [comment]
For the reference this is still reproducible... but also not very critical. I'm fine if we want to close this as WONTFIX and just let it be documented by the closed issue.

**Comment by RandomShaper** [comment]
I've not been able to allocate time for this. Feel free to close it. It's in my to do list, though, so it will be fixed one day. 😃

**Comment by akien-mga** [comment]
Yeah I think we can close it. It will be fixed as a side effect when we refactor PCK embedding, but the problem with `ld.gold` specifically is not worth keeping track of.
